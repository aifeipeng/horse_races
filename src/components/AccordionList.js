import React from 'react';
import {
    Accordion,
    AccordionItem,
    AccordionItemHeading,
    AccordionItemButton,
    AccordionItemPanel,
} from 'react-accessible-accordion';
import 'react-accessible-accordion/dist/fancy-example.css';

const AccordionList = (props) => {
    const { starts } = props

    return (
        <Accordion
            allowZeroExpanded={true}>
            {!!starts && starts.map(start => {
                return <AccordionItem
                    key={start.number}>
                    <AccordionItemHeading>
                        <AccordionItemButton>
                            <p>
                                {`Number: ${start.number}`}
                            </p>
                            <p>
                                {`Horse: ${start.horse.name}`}
                            </p>
                            <p>
                                {`Driver: ${start.driver.firstName} ${start.driver.lastName}`}
                            </p>
                        </AccordionItemButton>
                    </AccordionItemHeading>
                    <AccordionItemPanel>
                        <p>
                            {`Trainer: ${start.horse.trainer.firstName} ${start.horse.trainer.lastName}`}
                        </p>
                        <p>
                            {`Horses father: ${start.horse.pedigree.father.name}`}
                        </p>
                    </AccordionItemPanel>
                </AccordionItem>
            })}
        </Accordion>
    )
}


export default AccordionList
