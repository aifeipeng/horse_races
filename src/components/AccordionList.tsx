import React from 'react';
import {
    Accordion,
    AccordionItem,
    AccordionItemHeading,
    AccordionItemButton,
    AccordionItemPanel,
} from 'react-accessible-accordion';
import 'react-accessible-accordion/dist/fancy-example.css';

interface Start {
    number: string,
    driver: {
        firstName: string,
        lastName: string
    },
    horse: {
        name: string,

        trainer: {
            firstName: string,
            lastName: string
        },
        pedigree: { father: { name: string } }
    },
    startTime: string
}

const AccordionList = (props: { starts: [] }) => {
    const { starts } = props

    return (
        <Accordion
            allowZeroExpanded={true}>
            {!!starts && starts.map((start: Start) => {
                return <AccordionItem
                    key={start.number}>
                    <AccordionItemHeading>
                        <AccordionItemButton>
                            <p>
                                {`Number: ${start.number}`}
                            </p>
                            <p>
                                {`Horse: ${start.horse.name}`}
                            </p>
                            <p>
                                {`Driver: ${start.driver.firstName} ${start.driver.lastName}`}
                            </p>
                        </AccordionItemButton>
                    </AccordionItemHeading>
                    <AccordionItemPanel>
                        <p>
                            {`Trainer: ${start.horse.trainer.firstName} ${start.horse.trainer.lastName}`}
                        </p>
                        <p>
                            {`Horses father: ${start.horse.pedigree.father.name}`}
                        </p>
                    </AccordionItemPanel>
                </AccordionItem>
            })}
        </Accordion>
    )
}


export default AccordionList
