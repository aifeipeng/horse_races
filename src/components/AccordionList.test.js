import React from 'react';
import ReactDOM from 'react-dom';
import AccordionList from './AccordionList.tsx';
import ShallowRenderer from 'react-test-renderer/shallow';


const starts = [{
  number: "string",
  driver: {
      firstName: "string",
      lastName: "string"
  },
  horse: {
      name: "string",

      trainer: {
          firstName: "string",
          lastName: "string"
      },
      pedigree: { father: { name: "string" } }
  },
  startTime: "string"
},
{
  number: "string",
  driver: {
      firstName: "string",
      lastName: "string"
  },
  horse: {
      name: "string",

      trainer: {
          firstName: "string",
          lastName: "string"
      },
      pedigree: { father: { name: "string" } }
  },
  startTime: "string"
}
]

it('renders correctly', () => {
  const renderer = new ShallowRenderer()
  const tree = renderer
    .render(<AccordionList 
      starts={starts} />)
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccordionList />, div);
  ReactDOM.unmountComponentAtNode(div);
});
