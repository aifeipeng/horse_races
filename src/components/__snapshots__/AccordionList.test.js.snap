// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Accordion
  allowZeroExpanded={true}
  className="accordion"
>
  <AccordionItem
    className="accordion__item"
  >
    <AccordionItemHeading>
      <AccordionItemButton>
        <p>
          Number: string
        </p>
        <p>
          Horse: string
        </p>
        <p>
          Driver: string string
        </p>
      </AccordionItemButton>
    </AccordionItemHeading>
    <AccordionItemPanel
      className="accordion__panel"
    >
      <p>
        Trainer: string string
      </p>
      <p>
        Horses father: string
      </p>
    </AccordionItemPanel>
  </AccordionItem>
  <AccordionItem
    className="accordion__item"
  >
    <AccordionItemHeading>
      <AccordionItemButton>
        <p>
          Number: string
        </p>
        <p>
          Horse: string
        </p>
        <p>
          Driver: string string
        </p>
      </AccordionItemButton>
    </AccordionItemHeading>
    <AccordionItemPanel
      className="accordion__panel"
    >
      <p>
        Trainer: string string
      </p>
      <p>
        Horses father: string
      </p>
    </AccordionItemPanel>
  </AccordionItem>
</Accordion>
`;
